packages/openneuro-server/src/routes.ts

```diff
@@ -157,6 +164,58 @@ const routes = [
     middleware: [noCache, orcid.authCallback],
     handler: jwt.authSuccessHandler,
   },
+
+  // GitHub authentication route
+  {
+    method: "get",
+    url: "/auth/github",
+    middleware: [storeRedirect],
+    handler: (req, res, next) => {
+      const redirectTo = req.query.redirectTo || "/"
+      passport.authenticate("github", {
+        state: encodeURIComponent(redirectTo),
+      })(req, res, next)
+    },
+  },
+
+  {
+    method: "get",
+    url: "/auth/github/callback",
+    handler: (req, res, next) => {
+      passport.authenticate("github", (err, user, info) => {
+        const redirectTo = req.query.state
+          ? decodeURIComponent(req.query.state)
+          : "/"
+
+        // Remove any existing query parameters
+        const cleanRedirectTo = redirectTo.split("?")[0]
+
+        if (err) {
+          Sentry.captureException(err)
+          return res.redirect(
+            `${cleanRedirectTo}?error=${
+              encodeURIComponent(err.message || "github_auth_failed")
+            }`,
+          )
+        }
+
+        if (!user) {
+          Sentry.captureMessage(
+            `GitHub Auth Failed - Info: ${JSON.stringify(info)}`,
+            "warning",
+          )
+          return res.redirect(
+            `${cleanRedirectTo}?error=${
+              encodeURIComponent(info?.message || "github_auth_failed")
+            }`,
+          )
+        }
+
+        return res.redirect(`${cleanRedirectTo}?success=github_auth_success`)
+      })(req, res, next)
+    },
+  },
+
```

Comment:
The implementation here looks good, but I think these handlers should be imported from their own module, like the Google and ORCID ones.

Reply:
ok, those are moved to libs/authentication/github.ts